scripts/kconfig/kconfig.py

```diff
@@ -184,31 +185,26 @@ def promptless(sym):
 
     return not any(node.prompt for node in sym.nodes)
 
+
 def write_kconfig_filenames(paths, root_path, output_file_path):
     # 'paths' is a list of paths. The list has duplicates and the
     # paths are either absolute or relative to 'root_path'.
 
     # We need to write this list, in a format that CMake can easily
     # parse, to the output file at 'output_file_path'.
 
-    # The written list should also have absolute paths instead of
-    # relative paths, and it should not have duplicates.
+    # The written list has sorted normalized absolute paths, and it does not have duplicates.
+    # The list is sorted to be deterministic.
 
-    # Remove duplicates
-    paths_uniq = set(paths)
+    paths_uniq = sorted(set([normpath(abspath(join(root_path, path))) for path in paths]))
```

Comment:
Can use a generator to get rid of a temporary list (pair of `[]`). Think `realpath()` would work here too.

```python3
paths_uniq = sorted(set(realpath(join(root_path, path)) for path in paths))
```

Reply:
Can be simplified a bit more with a set comprehension:

```python
paths_uniq = sorted({realpath(join(root_path, path)) for path in paths})
```